### ISSUE TITLE
Add duplicate method to Dataframe

### ISSUE THREAD

**Issue by JulianCologne** [pola-rs/polars#7614]
### Problem description

sorry for bringing this up again (see similar issues: #6137 #5590), but I think this is a very common use case and would be a great addition to polars.

**Problem:
there is currently no easy/efficient/convenient way to keep all rows where a subset is duplicated while keeping all available columns!**

Imagine a simple use case where you have a list of users or customers containing duplicates and you want to export all rows where e.g. the firstname, lastname and email are duplicated and then compare/clean/combine the values of the remaining columns before merging.

# Examples

```python
data = {
    "first": ["Peter", "Peter", "Mary", "Mary", "David", "Olivia"],
    "last": ["Smith", "Smith", "Smith", "Smith", "Smith", "Smith"],
    "age": [30, 32, 16, 16, 61, None]
}
```

## Polars: Option 1

```python
(
    df
    #.drop_nulls()  # Problem!
    .filter(
        df.select('first', 'last').is_duplicated()
    )
)
```
Problems:
- does not work if size of df changes

## Polars: Option 2

```python
(
    df
    .groupby('first', 'last')
    .agg(
        pl.col('age')
    )
    .filter(
        pl.col('age').arr.lengths() > 1
    )
)
```
Problems:
- very verbose and not very readable
- should not be very efficient
- output not very readable if you want to compare the values of the duplicated rows

## Polars: Option 3

```python
(
    df
    .filter(
        pl.concat_str(['first', 'last']).is_duplicated()
    )
)
```
Problems:
- not very concise and efficient
- very inflexible (e.g. if you want to combine more than 2 columns or different data types)

## Pandas
```
df[df.duplicated(
    subset=['first', 'last'],
    keep=False,
)]
```
Perfect: concise, readable and flexible (with keep parameter)

# Suggested syntax

```python
df.duplicate(
    subset=subset,
    skip='first|last|none'  # analog to "unique" with keep='first|last|none'
)
```

**Comment by cmdlineluser** [comment]
`struct.is_duplicated` is pretty new - it was only added in [0.16.6](https://github.com/pola-rs/polars/releases/tag/py-0.16.6)

Perhaps some new struct examples that explicitly mention the word `subset` could be added to the documentation for better visibility?

The main result when searching the guide for struct seems to be:

https://pola-rs.github.io/polars-book/user-guide/dsl/custom_functions.html#combining-multiple-column-values

**Comment by ritchie46** [comment]
> I would think that no normal polars user and especially no polars beginner even knows about pl.struct because pandas does not have this concept

We don't aim to imitate pandas API. We aim to keep the available methods small and push users into the expression API.

> I myself try to avoid using pl.struct because last time I checked there was no documentation on this at all and I do not find this very intuitive to use

The `struct` is the idiomatic way to deal with multi-column expressions in polars. Please use them.

**Comment by JulianCologne** [comment]
@ritchie46 
> The `struct` is the idiomatic way to deal with multi-column expressions in polars. Please use them.

thanks for the tip, I will try.
It would be great if there would be some documentation, because the concept is already very different from eg pandas and a few examples would be really great! =)

> We don't aim to imitate pandas API. We aim to keep the available methods small and push users into the expression API.

I am absolutely with you there!
Polars is in my opinion much nicer to work with!

BUT, coming back to the issue, for me personally the API is inconsistent here.
For example, if I provide a `min` functionality, then I also expect `max`.
Likewise, if I provide a `unique` functionality, then I also expect `duplicate`.

As for using the `pl.struct` method, I think this should also be consistent for `unique` and `duplicate`.

Currently (inconsistent):
```python
# unique
df.unique(subset)

# duplicate
df.filter(pl.struct(subset).is_duplicated())
```

Option 1 (consistent):
```python
# unique
df.unique(subset)

# duplicate
df.duplicate(subset)
```

Option 2 (consistent, going with the `pl.struct` version!):
```python
# unique
df.filter(pl.struct(subset).is_unique())

# duplicate
df.filter(pl.struct(subset).is_duplicated())
```

Thinking about it, I think option 2 is actually really nice!
Maybe even consider deprecating `unique` on the DataFrame and only keep it on the Series to avoid questions like this in the future?

One addition I would like then be to add the `keep` parameter to `is_unique` otherwise we loose functionality.

**Comment by s-banach** [comment]
I'm +1 on adding `struct` examples to the documentation for basic functions like `unique()`. I was recently trying to do `concat_list(...).unique()` instead of `struct(...).unique()`, because I didn't know any better.

**Comment by cmdlineluser** [comment]
I'm not sure on what exactly the process is for adding documentation.

I've made an attempt here: https://gist.github.com/cmdlineluser/175cdb7f22d575b7c5797c717b69f0f5

**Comment by mcrumiller** [comment]
I almost think we should have a `pl.multicol` as an alias of `pl.struct.` This issue comes up all the time: I recall I wanted a multicolumn-level `n_unique` a while back and learned about the whole `df.select(pl.struct(...).unique()).unnest()` approach in the process. It's very long-winded to convert to struct, perform a multi-column operation, and unnest back to multi-column, and I think it would behoove polars to streamline this process a bit.

**Comment by cmdlineluser** [comment]
Yes, I had a similar thought with regards to a `pl.subset(...)` mirroring the `subset=` parameter found in the other functions. Wasn't sure if it was a good idea or not.

It seems some of the confusion comes from `pl.col` taking multiple column names.

`pl.col("a", "b").n_unique()`

People read it as the n_unique being applied to both columns as one "item" as opposed to it essentially being shorthand for 

`pl.col("a").n_unique(), pl.col("b").n_unique()`

So maybe something added to the [`.col()`](https://pola-rs.github.io/polars/py-polars/html/reference/expressions/api/polars.col.html#polars.col) docs could also be helpful.

**Comment by ion-elgreco** [comment]
> I almost think we should have a `pl.multicol` as an alias of `pl.struct.` This issue comes up all the time: I recall I wanted a multicolumn-level `n_unique` a while back and learned about the whole `df.select(pl.struct(...).unique()).unnest()` approach in the process. It's very long-winded to convert to struct, perform a multi-column operation, and unnest back to multi-column, and I think it would behoove polars to streamline this process a bit, maybe with an implicit `.unnest()` while retaining column names?

For me it took some time to understand when to use `pl.struct`, so some documentation or defining an alias for it will definitely help beginners.

**Comment by avimallu** [comment]
@ion-elgreco, that is being worked on. See https://github.com/pola-rs/polars-book/issues/329.
